### ISSUE TITLE
Flatpak applies older runtime extensions to newer runtime versions, causing breakage

### ISSUE THREAD

**Issue by elad661** [flatpak/flatpak#2291]
## Linux distribution and version
Fedora 29
## Flatpak version
Flatpak 1.0.4
## Description of the problem
If you have an app that uses `org.freedesktop.Platform/x86_64/18.08` and also have `org.freedesktop.Platform.GL.default/x86_64/1.6` installed, the extension (which is for an older version of the runtime) will be used even for apps that are based on the newer version.
## Steps to reproduce
1. Install `org.freedesktop.Platform.GL.default/x86_64/1.6` from the "gnome" remote (http://sdk.gnome.org/repo/)
2. Install `org.freedesktop.Platform/x86_64/18.08` from flathub
3. `flatpak run org.freedesktop.Platform/x86_64/18.08`
4. Note that you'll have `/usr/lib/x86_64-linux-gnu/GL/default` containing libraries coming from the older extension

This breaks some apps, like Spotify, because it will try to load the older GL drivers and fail.
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/issues/458

I think it runtime extensions should be coupled to a specific version of the runtime, and not apply on newer versions.

**Comment by nanonyme** [comment]
They are,  sort of. Does Spotify inherit extensions from base app?

**Comment by elad661** [comment]
@nanonyme as I've shown in the steps to reproduce, you don't even need spotify to see the problem - just running `flatpak run org.freedesktop.Platform/x86_64/18.08` is enough to see that it will have files that come from the old extension

**Comment by nanonyme** [comment]
Note also the usecase that nvidia GL extension doesn't follow runtime versioning

**Comment by nanonyme** [comment]
Seems the main problem here is extensions get loaded without explicit opt-in in manifest

**Comment by TingPing** [comment]
This is a bug in the 18.08 runtime. It was kinda fixed in this MR but never merged: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/merge_requests/458

**Comment by TingPing** [comment]
I made a new PR: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/merge_requests/671

**Comment by nanonyme** [comment]
Yeah, I wouldn't expect that 458 to be merged anytime soon, it was kinda experimental. I hope there aren't any more gold nuggets like this there, this is already the second.
